src/lib/components/drs/DrsMain.js

```diff
@@ -0,0 +1,312 @@
+    version: _.pick(formProps, ['version', 'setVersion']),
+    isBundle: _.pick(formProps, ['is_bundle', 'setIsBundle']),
+    mimeType: _.pick(formProps, ['mime_type', 'setMimeType']),
+    size: _.pick(formProps, ['size', 'setSize']),
+    aliases: _.pick(formProps, ['aliases', 'addAlias', 'setAlias', 'removeAlias']),
+    checksums: {..._.pick(formProps, ['checksums', 'addChecksum', 'setChecksumType', 'setChecksumChecksum', 'removeChecksum']), displayChecksumTypes: displayChecksumTypes},
+    children: _.pick(formProps, ['drs_object_children', 'addChild', 'setChildId', 'setChildName', 'setChildValid', 'setChildInvalid', 'unsetChildValidity', 'removeChild']),
+    parents: _.pick(formProps, ['drs_object_parents', 'addParent', 'setParentId', 'setParentName', 'setParentValid', 'setParentInvalid', 'unsetParentValidity', 'removeParent']),
+    fileAccessObjects: _.pick(formProps, ['file_access_objects', 'addFileAccessObject', 'setFileAccessObjectPath', 'removeFileAccessObject']),
+    awsS3AccessObjects: _.pick(formProps, ['aws_s3_access_objects', 'addAwsS3AccessObject', 'setAwsS3AccessObjectRegion', 'setAwsS3AccessObjectBucket', 'setAwsS3AccessObjectKey', 'removeAwsS3AccessObject']),
+    submit: {activeDrsObject: activeDrsObject, setSuccessMessage: setSuccessMessage, ..._.pick(formProps, ['retrieveDrsObjectsList'])}
+  }
+
+  /* ##################################################
+   * EFFECTS
+   * ################################################## */
+
+  // add the history listener, modifies the activeDrsObject based on the current
+  // URL path
+  let history = useHistory();
+  useEffect(() => {
+    history.listen((location, action) => {
+      let re = new RegExp('/drs/(.+)');
+      let match = location.pathname.match(re);
+      if (match) {
+        match[1] === 'new' ? activeDrsObjectFunctions.reset() : activeDrsObjectFunctions.retrieve(match[1]);
+      }
+    })
+  }, [])
+
+  // initialize the drsObjectList upon first load
+  useEffect(() => {
+    retrieveDrsObjectsList();
+  }, []);
+
+  // update the allowed checksum types list every time any checksum type
+  // changes
+  useEffect(() => {
+    let transientActiveDrsObject = {...activeDrsObject};
+    let transientDisplayChecksumTypes = emptyDisplayChecksumTypes();
+    let checksumTypesEncountered = [];
+    let unsetChecksumTypeEncountered = false;
+    
+    transientActiveDrsObject.checksums.forEach((checksum, index) => {
+      if (checksum.type !== '') {
+        if (checksumTypesEncountered.includes(checksum.type)) {
+          unsetChecksumTypeEncountered = true;
+          checksum.type = '';
+        } else {
+          transientDisplayChecksumTypes[checksum.type] = index;
+          checksumTypesEncountered.push(checksum.type);
+        }
+      }
+    })
+    setDisplayChecksumTypes(transientDisplayChecksumTypes);
+
+    if (unsetChecksumTypeEncountered) {
+      setActiveDrsObject(transientActiveDrsObject);
+    }
+  }, [activeDrsObject]);
+
```

Comment:
The issue with displaying and hiding the correct Checksum Types still seems to be present. 
For example, if all three Checksum instances have the Type field populated, if the user then selects the Type dropdown for the first Checksum instance, all three Type options can be selected. Additionally, if the user selects one of these options (which has already been selected for another Checksum instance), the user can select it successfully, resulting in the Type field being populated, and the Type field of the Checksum that originally held that value being cleared.